src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -792,6 +792,8 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger? errorLogger, Cancella
 
             cancellationToken.ThrowIfCancellationRequested();
 
+            Console.OutputEncoding = Encoding.UTF8;
```

Comment:
A bit skeptical that we can make this change. Consider that this is already controlled via the MSBuild property `$(Utf8Output)`. You can see the compiler taking advantage of it here 

https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/MSBuildTask/ManagedCompiler.cs#L452

This is very old legacy behavior and to be honest I don't know why it's done that way. Hence hard to predict the compat problems this could or could not create if we effectively begin ignoring the output for the console. 

@rainersigwald may have some ideas though. If he thinks it's safe then we can consider taking this forward.

Reply:
The compiler never writes to error for reasons I can't actually explain.

Reply:
This is all a mess for sure. However, the desired future state is pretty clear IMO: every command line tool should emit UTF-8 and MSBuild should happily deal with that.

I think I'd be comfortable with the Roslyn MSBuild tasks ignoring the `Utf8Output` argument and always setting it to `true`. Looking through MSBuild it looks like that's a Roslyn property (not used other than in clones of `CoreCompile`).

> hard to predict the compat problems this could or could not create if we effectively begin ignoring the output for the console.

This is definitely true. For the compiler itself invoked from outside of MSBuild I don't know of a good safe option: if you just do the default, that can't represent everything and is an internationalization problem. If you suddenly switch to unconditionally emitting UTF-8, that's The Right Thing™️ but probably breaks tools that expect `$current_os_codepage`.

Reply:
@jaredpar Any suggestions on how we can move forward here? or should we close the PR and the linked issue?

Reply:
I think my preference at this point would be to close. The reason being that generally speaking invoking csc directly is not a primary scenario for newer code. It is always wrapped in some msbuild invocation which by default is going to do the right thing. The only code bases that continue to invoke csc directly are the ones that are generally the most compat sensitive.